src/main/java/dk/bec/polonez/reservationsystem/service/MailService.java

```diff
@@ -0,0 +1,60 @@
+package dk.bec.polonez.reservationsystem.service;
+
+import dk.bec.polonez.reservationsystem.dto.notificationDto.ReservationConfirmationInput;
+import dk.bec.polonez.reservationsystem.dto.notificationDto.ReservationStatusNotificationInput;
+import dk.bec.polonez.reservationsystem.dto.notificationDto.UpcomingEventsNotificationInput;
+import dk.bec.polonez.reservationsystem.model.NotificationTemplate;
+import dk.bec.polonez.reservationsystem.repository.NotificationTemplateRepository;
+import lombok.RequiredArgsConstructor;
+import org.springframework.mail.SimpleMailMessage;
+import org.springframework.mail.javamail.JavaMailSender;
+import org.springframework.stereotype.Service;
+
+@Service
+@RequiredArgsConstructor
+public class MailService {
+
+    private final JavaMailSender mailSender;
+
+    private final NotificationTemplateRepository templateRepository;
+
+    public void sendReservationStatusNotification(String emailAddress, ReservationStatusNotificationInput input) {
+        NotificationTemplate template = templateRepository.getByName("RESERVATION_STATUS");
```

Comment:
name do stałej, albo jakis enum

na pewno nie powinno to byc hard coded

podobnie w innych metodach

Reply:
Enum is ideal IMO.